homeassistant/components/aten_pe/switch.py

```diff
@@ -0,0 +1,118 @@
+"""The ATEN PE switch component."""
+
+import logging
+
+from atenpdu import AtenPE
+import voluptuous as vol
+
+from homeassistant.components.switch import (
+    DEVICE_CLASS_OUTLET,
+    PLATFORM_SCHEMA,
+    SwitchDevice,
+)
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_USERNAME
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_AUTH_KEY = "auth_key"
+CONF_COMMUNITY = "community"
+CONF_PRIV_KEY = "priv_key"
+DEFAULT_COMMUNITY = "private"
+DEFAULT_PORT = "161"
+DEFAULT_USERNAME = "administrator"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_COMMUNITY, default=DEFAULT_COMMUNITY): cv.string,
+        vol.Optional(CONF_USERNAME, default=DEFAULT_USERNAME): cv.string,
+        vol.Optional(CONF_AUTH_KEY): cv.string,
+        vol.Optional(CONF_PRIV_KEY): cv.string,
+    }
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the ATEN PE switch."""
+    dev = AtenPE(
+        node=config[CONF_HOST],
+        serv=config[CONF_PORT],
+        community=config[CONF_COMMUNITY],
+        username=config[CONF_USERNAME],
+        authkey=config.get(CONF_AUTH_KEY),
+        privkey=config.get(CONF_PRIV_KEY),
+    )
+
+    try:
+        await hass.async_add_executor_job(dev.loadMibs)
+
+        mac = await dev.deviceMAC()
+
+        switches = []
+        async for outlet in dev.outlets():
+            switches.append(AtenSwitch(dev, mac, outlet.id, outlet.name))
+    except Exception:  # pylint: disable=broad-except
```

Comment:
Please only wrap the line(s) that can raise with the try... except block.

Reply:
Please catch one or more specific exceptions. We want to avoid catching broad exceptions if possible.

What exceptions do we expect? Preferably the library should raise its own exceptions.

Reply:
I modified the library to raise its own exceptions and updated the code above accordingly.